samples/PixelShaders/Retro.hlsl

```diff
@@ -2,47 +2,46 @@
 Texture2D shaderTexture;
 SamplerState samplerState;
 
-cbuffer PixelShaderSettings {
-  float  Time;
-  float  Scale;
-  float2 Resolution;
-  float4 Background;
+cbuffer PixelShaderSettings
+{
+    float time;
+    float scale;
+    float2 resolution;
+    float4 background;
 };
 
-#define SCANLINE_FACTOR 0.5
-#define SCALED_SCANLINE_PERIOD Scale
-#define SCALED_GAUSSIAN_SIGMA (2.0*Scale)
+#define SCANLINE_FACTOR 0.5f
+#define SCALED_SCANLINE_PERIOD scale
+#define SCALED_GAUSSIAN_SIGMA (2.0f * scale)
 
 static const float M_PI = 3.14159265f;
 
 float Gaussian2D(float x, float y, float sigma)
 {
-    return 1/(sigma*sqrt(2*M_PI)) * exp(-0.5*(x*x + y*y)/sigma/sigma);
+    return 1 / (sigma * sqrt(2 * M_PI)) * exp(-0.5 * (x * x + y * y) / sigma / sigma);
 }
 
 float4 Blur(Texture2D input, float2 tex_coord, float sigma)
 {
-    uint width, height;
+    float width, height;
     shaderTexture.GetDimensions(width, height);
 
-    float texelWidth = 1.0f/width;
-    float texelHeight = 1.0f/height;
+    float texelWidth = 1.0f / width;
+    float texelHeight = 1.0f / height;
 
     float4 color = { 0, 0, 0, 0 };
 
-    int sampleCount = 13;
+    float sampleCount = 13;
 
-    for (int x = 0; x < sampleCount; x++)
+    for (float x = 0; x < sampleCount; x++)
     {
         float2 samplePos = { 0, 0 };
+        samplePos.x = tex_coord.x + (x - sampleCount / 2.0f) * texelWidth;
 
-        samplePos.x = tex_coord.x + (x - sampleCount/2) * texelWidth;
-        for (int y = 0; y < sampleCount; y++)
+        for (float y = 0; y < sampleCount; y++)
         {
-            samplePos.y = tex_coord.y + (y - sampleCount/2) * texelHeight;
-            if (samplePos.x <= 0 || samplePos.y <= 0 || samplePos.x >= width || samplePos.y >= height) continue;
-
-            color += input.Sample(samplerState, samplePos) * Gaussian2D((x - sampleCount/2), (y - sampleCount/2), sigma);
+            samplePos.y = tex_coord.y + (y - sampleCount / 2.0f) * texelHeight;
+            color += input.Sample(samplerState, samplePos) * Gaussian2D(x - sampleCount / 2.0f, y - sampleCount / 2.0f, sigma);

```

Comment:
FYI with `D3D11_TEXTURE_ADDRESS_BORDER` we already ensure that `Sample()` returns `float4(0, 0, 0, 0)`. This removes the need for the if condition, which is one of the things that produces a warning by default (and for a good reason - loops like that are impossible on GPUs).